include/llvm-dialects/Dialect/Dialect.td

```diff
@@ -279,6 +279,21 @@ def : AttrLlvmType<AttrI16, I16>;
 def : AttrLlvmType<AttrI32, I32>;
 def : AttrLlvmType<AttrI64, I64>;
 
+def ImmutableStringAttr : Attr<"::llvm::StringRef"> {
+  let toLlvmValue = [{ $_builder.CreateGlobalString($0) }];
+  let fromLlvmValue = [{ ::llvm::cast<::llvm::ConstantDataArray>(::llvm::cast<::llvm::GlobalVariable>($0)->getInitializer())->getAsString() }];
+  let isImmutable = "true";
+}
+ 
+def ImmutableStringType : BuiltinType {
+  let arguments = (args type:$self);
+  let evaluate = "::llvm::PointerType::get($_context, 0)";
+  let check = "$self->isPointerTy()";
+  let capture = [];
+}
+ 
+def : AttrLlvmType<ImmutableStringAttr, ImmutableStringType>;
```

Comment:
I would have thought this could just use `Ptr` instead of a new type. Does that not work?